UnityProject/Assets/Scripts/Items/Cards/Emag.cs

```diff
@@ -0,0 +1,123 @@
+﻿using System.Collections;
+using System.Collections.Generic;
+using UnityEngine;
+using Mirror;
+
+/// <summary>
+/// Emag charges handler
+/// </summary>
+public class Emag : NetworkBehaviour, IServerSpawn
+{
+	private SpriteHandler spriteHandler;
+
+    [Tooltip("Number of charges emags start with")]
+    [SerializeField]
+    public int startCharges = 3;
+
+    [Tooltip("Number of seconds it takes to regenerate 1 charge")]
+    [SerializeField]
+    public float rechargeTimeInSeconds = 10f;
+
+    [SyncVar(hook=nameof(SyncCharges))]
+    private int charges;
+
+	/// <summary>
+	/// Number of charges left on emag
+	/// </summary>
+    public int Charges => charges;
+
+    private string OutOfChargesSFX = "Sparks04";
+
+    #region SyncVarFuncs
+        void Awake()
+        {
+            EnsureInit();
+        }
+
+        private void EnsureInit()
+        {
+            if(spriteHandler == null)
+            {
+                charges = startCharges;
+                spriteHandler = gameObject.transform.Find("Charges").GetComponent<SpriteHandler>();
+            }
+        }
+
+        public void OnDisable()
+        {
+            if (isServer)
+            {
+                UpdateManager.Remove(CallbackType.PERIODIC_UPDATE, RegenerateCharge);
+            }
+        }
+
+        public override void OnStartClient()
+        {
+            EnsureInit();
+            SyncCharges(Charges, charges);
+            base.OnStartClient();
+        }
+        
+        public void OnSpawnServer(SpawnInfo info)
+        {
+            SyncCharges(startCharges, startCharges);
+        }
+    #endregion
+
+    private void SyncCharges(int oldCharges, int newCharges)
```

Comment:
Codacy found an issue: [Remove this unused method parameter 'oldCharges'.](https://app.codacy.com/gh/unitystation/unitystation/pullRequest?prid=6380756)

Reply:
You can ignore this suggestion when it is a `SyncVar`.